feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailViewModel.kt

```diff
@@ -23,17 +26,96 @@ class MovieDetailViewModel @Inject constructor(
   savedStateHandle: SavedStateHandle,
   private val getMovieDetail: GetMovieDetailsUseCase,
   private val getStreamInfo: GetStreamInfoUseCase,
+  private val getCredit: GetMovieCreditUseCase,
+  private val getSimilar: GetSimilarUseCase,
   private val traktHistoryRepository: TraktHistoryRepository,
 ) : ViewModel() {
 
   private val videoId: Int = savedStateHandle["video_id"] ?: throw IllegalStateException("videoId is required")
   private val _state: MutableStateFlow<MovieDetailState> = MutableStateFlow(MovieDetailState())
   val state = _state.asStateFlow()
+  private val _similarState: MutableStateFlow<MovieDetailSimilarState> = MutableStateFlow(MovieDetailSimilarState())
+  val similarState = _similarState.asStateFlow()
+
+  private val _creditState: MutableStateFlow<MovieDetailCreditState> = MutableStateFlow(MovieDetailCreditState())
+  val creditState = _creditState.asStateFlow()
 
   val navigateToPlayer = MutableSharedFlow<String?>()
 
   init {
     load()
+    loadSimilar()
+    loadCredits()
+  }
+
+  private fun loadSimilar() = viewModelScope.launch {
+    _similarState.value = _similarState.value.copy(isLoading = true)
+    when (val result = getSimilar(videoId)) {
+      is Failure -> {
+        when (result.error) {
+          is GeneralError.ApiError -> {
+            _state.value = _state.value.copy(
+              error = result.error,
+              message = (result.error as GeneralError.ApiError).message,
+              isLoading = false,
+            )
```

Comment:
I highly recommend using `update` function instead of assigning new value to `value`:

```suggestion
            _state.update { state ->
                 state.copy(
                     error = result.error,
                     message = (result.error as GeneralError.ApiError).message,
                     isLoading = false,
                 )
             }
```

Reply:
I changed my code and also changed everywhere that we used this kind of assigning method